base/multidimensional.jl

```diff
@@ -723,14 +712,14 @@ function accumulate_pairwise!(op::Op, result::AbstractVector, v::AbstractVector)
     n = length(li)
     n == 0 && return result
     i1 = first(li)
-    @inbounds result[i1] = v1 = v[i1]
+    @inbounds result[i1] = v1 = reduce_first(op,v[i1])
     n == 1 && return result
     _accumulate_pairwise!(op, result, v, v1, i1+1, n-1)
     return result
 end
 
 function accumulate_pairwise(op, v::AbstractVector{T}) where T
-    out = similar(v, rcum_promote_type(op, T))
+    out = similar(v, promote_op(op, T))
```

Comment:
This (and others) should probably be `promote_op(op, T, T)`: though built-in operators provide unary versions, anonymous functions typically won't.

Reply:
Oh thanks! I somehow assumed `promote_op(op,T) = promote_op(op,T,T)` without checking. Documenting it is a good idea!